src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+internal sealed class AsyncToSyncMethodTransformer : SyntaxTransformer {
+	public AsyncToSyncMethodTransformer( SemanticModel model ) : base( model ) { }
+
+	public TransformResult<MethodDeclarationSyntax> Transform( MethodDeclarationSyntax decl ) {
+		// TODO: remove CancellationToken parameters
+		decl = decl.WithAttributeLists( ReplaceGenerateSyncAttribute( decl.AttributeLists ) )
+			.WithModifiers( RemoveAsyncModifier( decl.Modifiers ) )
+			.WithIdentifier( RemoveAsyncSuffix( decl.Identifier ) )
+			.WithReturnType( TransformReturnType( decl.ReturnType ) )
+			.WithExpressionBody( Transform( decl.ExpressionBody ) )
+			.WithBody( Transform( decl.Body ) );
+
+		return GetResult( decl );
+	}
+
+	private SyntaxList<AttributeListSyntax> ReplaceGenerateSyncAttribute(
+		SyntaxList<AttributeListSyntax> attributeLists
+	) => TransformAll( attributeLists, ReplaceGenerateSyncAttribute );
+
+	private AttributeListSyntax? ReplaceGenerateSyncAttribute(
+		AttributeListSyntax attributeList
+	) => attributeList.WithAttributes(
+		TransformAll( attributeList.Attributes, ReplaceGenerateSyncAttribute )
+	);
+
+	private AttributeSyntax ReplaceGenerateSyncAttribute(
+		AttributeSyntax attribute
+	) {
+		// Attributes other than [GenerateSync] pass through
+		if( !SyncGenerator.IsGenerateSyncAttribute( m_model, attribute ) ) {
+			return attribute;
+		}

```

Comment:
Usage of the SemanticModel. This one can maybe go away but it isn't the only usage.

Reply:
SyncGenerator hasn't been introduced yet.

Reply:
Yeah that's why this is draft--just working out the kinks before adding reviewers. 😄